xla/stream_executor/gpu/gpu_command_buffer.cc

```diff
@@ -571,7 +572,18 @@ absl::Status GpuCommandBuffer::Finalize() {
   // Maybe dump created GPU graph to a dot file for debugging.
   if (state_ == State::kCreate &&
       (VLOG_IS_ON(10) || (VLOG_IS_ON(9) && mode_ == Mode::kPrimary))) {
-    std::string path = tsl::io::GetTempFilename(/*extension=*/"dot");
+    std::string path;
+    std::string dump_dir;
+    if (!tsl::ReadStringFromEnvVar("COMMAND_BUFFER_DUMP_DIR", "", &dump_dir)
```

Comment:
xla already provides a convention and flag for dumping debug data 
```
--xla_dump_to=""  string  A directory to dump xla debug data to.
```

@ezhulenev Do you think we need to introduce separate environment variable COMMAND_BUFFER_DUMP_DIR

Reply:
before_validation.ttir.txt dump example
[xla/backends/gpu/codegen/triton/fusion_emitter.cc](https://github.com/openxla/xla/blob/main/xla/backends/gpu/codegen/triton/fusion_emitter.cc#L1980)

```
// dump.h
// Consolidated utilities for logging information during compilation, usually
// based on the options specified in the DebugOptions proto.
//
// Most functions here take an HloModule and read the DebugOptions from the
// module's config.
```
[xla/service/dump.h](https://github.com/openxla/xla/blob/main/xla/service/dump.h)